Documentation/Testing/ExtensionTesting.rst

```diff
@@ -2,846 +2,106 @@
+.. _testing-extensions-linting:
 
-You may now see something similar to this:
+Linting
+=======
 
-.. code-block:: text
+A linting test ensures that the syntax of the language used is correct.
 
-    Creating network "local_default" with the default driver
-    PHP 8.2.6 (cli) (built: May 13 2023 01:04:28) (NTS)
-    PHPUnit 10.2.0 by Sebastian Bergmann and contributors.
+In TYPO3 extensions commonly the following formats are linted:
 
-    Runtime:       PHP 8.2.6
-    Configuration: /var/www/mysite/typo3conf/ext/styleguide/Build/UnitTests.xml
+*   PHP in the supported versions
+*   TypoScript
+*   YAML
+*   JavaScript / TypeScript
 
-    .                                                                   1 / 1 (100%)
+And other programming and configuration languages used.
 
-    Time: 00:00.007, Memory: 12.00 MB
+..  todo: Add links to examples from blog extension and maybe others here
 
-    OK (1 test, 5 assertions)
-    Removing local_unit_run_1f529b0fb49d ... done
-    Removing network local_default
+.. _testing-extensions-cgl:
 
+Coding guidelines (CGL)
+=======================
 
-If there is no test output, try changing the verbosity when you run runTests.sh:
+If more then one person is working on code, coding guideline are a must-have.
+No matter which tool you use or which rules you choose to apply, it is important
+that the rules can be applied automatically.
 
-.. code-block:: shell
+..  todo: Add links to examples from blog extension and maybe others here
 
-   enetcache> Build/Scripts/runTests.sh -v
+.. _testing-extensions-static:
 
-Use -h to see all options:
+Static code analysis
+====================
 
-.. code-block:: shell
+Static code analysis tools are highly recommended to be used with PHP. The most
+common tools used are PHPStan and Psalm. No matter the tool you use or the
+rules and levels you apply: You should use one.
 
-   enetcache> Build/Scripts/runTests.sh -h
+There are also static code analysis tools for TypeScript and JavaScript.
 
-On some versions of MacOS you might get the following error message when executing :file:`runTests.sh`:
+..  todo: Link to a detailed chapter on static code analysis and to examples
 
-.. code-block:: shell
+.. _testing-extensions-unit:
 
-    $ ./Build/Scripts/runTests.sh
-    readlink: illegal option -- f
-    usage: readlink [-n] [file ...]
-    Creating network "local_default" with the default driver
-    ERROR: Cannot create container for service unit: invalid volume specification: '.:.:rw':
-    invalid mount config for type "volume": invalid mount path: '.' mount path must be absolute
-    Removing network local_default
+Unit tests
+==========
 
-To solve this issue follow the steps described `here <https://biercoff.com/fixing-readlink-illegal-option-f-error-on-a-mac/>`_ to install greadlink which supports the needed --f option.
+Unit tests are executing the code to be tested. They are run on an isolated
+class or method. All relationships are mocked.
```

Comment:
`All relationships are mocked`

I would rephrase that. I guess, we should phrase this in a way which emphasizes that relations and services (DB Calls, Api/Curl calls **must** be mocked in the test and that no full instance setup is available (DI, Database, Config, Bootstrap, etc ...).

Maybe we can add a short mantra (quote) for it ?

```rst
..  info::

    Rule of dumb: Unit tests are isolatetd tests not using real services.
```

Reply:
I tried to apply your changes please have another look